x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/strategies/merge_all_fields_with_source.test.ts

```diff
@@ -1523,14 +1525,113 @@ describe('merge_all_fields_with_source', () => {
       const fields: SignalSourceHit['fields'] = {
         foo: [{ bar: ['single_value'], zed: ['single_value'] }],
       };
-      const doc: SignalSourceHit = { ...emptyEsResult(), _source, fields };
+      const doc: SignalSourceHit = { ...emptyEsResult(), _source: cloneDeep(_source), fields };
       const merged = mergeAllFieldsWithSource({ doc, ignoreFields: [] })._source;
       expect(merged).toEqual<ReturnTypeMergeFieldsWithSource>({
         foo: [{ bar: ['single_value'], zed: ['single_value'] }],
       });
     });
   });
 
+  describe('mixed flattened and nested fields for the _source', () => {
+    test('merges fields into source if it is empty', () => {
+      const _source: SignalSourceHit['_source'] = {
+        'email.headers': {},
+      };
+
+      const fields: SignalSourceHit['fields'] = {
+        'email.headers.x-test': ['from fields'],
+      };
+      const doc: SignalSourceHit = { ...emptyEsResult(), _source: cloneDeep(_source), fields };
+      const merged = mergeAllFieldsWithSource({ doc, ignoreFields: [] })._source;
+      expect(merged).toEqual<ReturnTypeMergeFieldsWithSource>({
+        email: {
+          headers: {
+            'x-test': 'from fields',
+          },
+        },
+        // there is leftover of the empty flattened object left out, since we can't create wildcard
+        // map of string based paths to array paths in buildFieldsKeyAsArrayMap
+        'email.headers': {},
+      });
+    });
+
+    test('merges fields into source if it is "undefined"', () => {
+      const _source: SignalSourceHit['_source'] = {
+        'email.headers': { 'x-test': undefined },
+      };
+
+      const fields: SignalSourceHit['fields'] = {
+        'email.headers.x-test': ['from fields'],
+      };
+      const doc: SignalSourceHit = { ...emptyEsResult(), _source: cloneDeep(_source), fields };
+      const merged = mergeAllFieldsWithSource({ doc, ignoreFields: [] })._source;
+      expect(merged).toEqual<ReturnTypeMergeFieldsWithSource>({
+        'email.headers': {
+          'x-test': 'from fields',
+        },
+      });
+    });
+
+    test('merges and unboxes primitive fields into source', () => {
```

Comment:
Is this maybe two tests? 
1. Favors values from the fields response, and 
2. Unwraps a single-element array into the value of the first element

Also nit: I'm not sure if "unbox" is a well-understood term in javascript; better to be explicit, if possible.

Reply:
It feels redundant to have 2 tests doing the same, as we only do one assertion anyway here. There is the one only property that is tested, after it has been merged. It also lies in line with the rest test [coverage](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/strategies/merge_all_fields_with_source.test.ts#L374)
How would you see having 2 tests here and avoiding unnecessary code duplication?


"unbox" is something taken from other tests in similar cases 
https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/strategies/merge_all_fields_with_source.test.ts#L374
+
as well as from merge strategy README: https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/source_fields_merging/README.md